src/app/plugins/help.plugin.ts

```diff
@@ -1,38 +1,39 @@
 import { Plugin } from '../../common/plugin';
-import { IContainer, IMessage, ChannelType } from '../../common/types';
+import { IContainer, IMessage, IRichEmbed, ChannelType } from '../../common/types';
 import Constants from '../../common/constants';
+import { RichEmbed } from 'discord.js';
 
 export class HelpPlugin extends Plugin {
   public name: string = 'Help Plugin';
   public description: string = 'Displays supported commands and usage statements.';
   public usage: string = 'help';
   public permission: ChannelType = ChannelType.Bot;
-  private _commandsResponse: string = '';
+  private _embed: IRichEmbed = new RichEmbed;
   constructor(public container: IContainer) {
     super();
   }
 
   public async execute(message: IMessage, args?: string[]) {
-    if (this._commandsResponse.length === 0) {
-      this._generateCommandsResponse();
+    if (this._embed.length === 0) {
```

Comment:
idk if checking the length of an object is supposed to work like correctly or with this syntax, but it works fine when I tested it. Not sure if this is what it should be doe

Reply:
You would probably want to be checking the length of the `fields` property which should contain an array of the fields within the RichEmbed object